Lib/http/cookiejar.py

```diff
@@ -448,7 +448,7 @@ def join_header_words(lists):
         attr = []
         for k, v in pairs:
             if v is not None:
-                if not re.search(r"^\w+$", v):
+                if not v.isalnum() and '_' not in v:
```

Comment:
This is not correct. For example, it produces incorrect results for "foo;_" and "foo_bar".

Reply:
Wait. How did I miss that one? My bad for this oversight.

Reply:
Ah . I think I messed up my boolean logic...